gradio/external.py

```diff
@@ -584,12 +586,114 @@ def fn(*data):
     return interface
 
 
+TEXT_FILE_EXTENSIONS = (
+    ".doc",
+    ".docx",
+    ".rtf",
+    ".epub",
+    ".odt",
+    ".odp",
+    ".pptx",
+    ".txt",
+    ".md",
+    ".py",
+    ".ipynb",
+    ".js",
+    ".jsx",
+    ".html",
+    ".css",
+    ".java",
+    ".cs",
+    ".php",
+    ".c",
+    ".cc",
+    ".cpp",
+    ".cxx",
+    ".cts",
+    ".h",
+    ".hh",
+    ".hpp",
+    ".rs",
+    ".R",
+    ".Rmd",
+    ".swift",
+    ".go",
+    ".rb",
+    ".kt",
+    ".kts",
+    ".ts",
+    ".tsx",
+    ".m",
+    ".mm",
+    ".mts",
+    ".scala",
+    ".dart",
+    ".lua",
+    ".pl",
+    ".pm",
+    ".t",
+    ".sh",
+    ".bash",
+    ".zsh",
+    ".bat",
+    ".coffee",
+    ".csv",
+    ".log",
+    ".ini",
+    ".cfg",
+    ".config",
+    ".json",
+    ".proto",
+    ".yaml",
+    ".yml",
+    ".toml",
+    ".sql",
+)
+IMAGE_FILE_EXTENSIONS = (".png", ".jpg", ".jpeg", ".gif")
+
+
+def format_prompt(message: str | MultimodalValue) -> list[dict]:
+    if isinstance(message, str):
+        text = message
+        files = []
+    else:
+        text = message.get("text", None)
+        files = message.get("files", [])
+    image_files, text_files = [], []
+    for file in files:
+        if file.lower().endswith(TEXT_FILE_EXTENSIONS):
+            text_files.append(file)
+        else:
+            image_files.append(file)
+    content = []
+    if text or text_files:
+        text = text or ""
+        text += "\n".join(
+            [f"\n## {Path(file).name}\n{Path(file).read_text()}" for file in text_files]
+        )
+        content.append({"type": "text", "text": text})
+    if image_files:
+        content.extend(
+            [
+                {
+                    "type": "image_url",
+                    "image_url": {"url": encode_url_or_file_to_base64(image)},
+                }
+                for image in image_files
+            ]
+        )
+    return content
+
+
 @document()
 def load_chat(
     base_url: str,
     model: str,
     token: str | None = None,
     *,
+    file_types: Literal["text_files", "images"]
```

Comment:
Not intuitive that `text_files` includes any text-encoded files, I would have just expected .txt files. Perhaps "text_encoded" is better. And I would rename "images" -> "image" as it is consistent with how one specifies images in `file_types` param in other components